program/src/instructions/shared_accounts.rs

```diff
@@ -0,0 +1,235 @@
+use anchor_lang::prelude::*;
+use program::AmmProgram;
+use tensor_toolbox::{escrow, shard_num};
+use whitelist_program::WhitelistV2;
+
+use crate::{error::ErrorCode, *};
+
+use super::constants::TFEE_PROGRAM_ID;
+
+/* AMM Protocol shared account structs*/
+
+/// Shared accounts for transfer instructions: deposit & withdraw
+/// Mint and token accounts are not included here as the AMM program supports multiple types of
+/// NFTs, not all of which are SPL token based.
+#[derive(Accounts)]
+pub struct TransferShared<'info> {
+    /// The owner of the pool and the NFT.
+    #[account(mut)]
+    pub owner: Signer<'info>,
+
+    /// The pool the asset is being transferred to/from.
+    #[account(
+        mut,
+        seeds = [
+            b"pool",
+            owner.key().as_ref(),
+            pool.pool_id.as_ref(),
+        ],
+        bump = pool.bump[0],
+        has_one = owner @ ErrorCode::BadOwner,
+        // can only trasnfer to/from NFT & Trade pools
+        constraint = pool.config.pool_type == PoolType::NFT || pool.config.pool_type == PoolType::Trade @ ErrorCode::WrongPoolType,
+        constraint = pool.expiry >= Clock::get()?.unix_timestamp @ ErrorCode::ExpiredPool,
```

Comment:
Transfers don't care about expiry, right? Otherwise how does an owner withdraw their own NFTs?

Reply:
Yep, you're right.